### PR TITLE
Add warning to RemoveString comment

### PR DESCRIPTION
**Pull request by GerardRodes**
I noticed this and I think that it is important to explain it on the comment so we can avoid future bugs

### CODE REVIEW

File: sliceutil/sliceutil.go [hunk 1]
```diff
@@ -160,7 +160,9 @@ func FilterString(list []string, fun func(string) bool) []string {
 	return ret
 }
 
-// RemoveString removes any occurrence of a string from a list.
+// RemoveString removes any occurrence of a string from a slice.
+// It mutates the underlaying array and returns a new slice containing
```
Comment (ripexz) [inline sliceutil/sliceutil.go:164]: Hmm, the whole mutation of the underlying is a bit dodgy though, maybe better to clone it... Integrity comes before efficiency? 😅

Comment (rafaeljusto) [reply sliceutil/sliceutil.go:164]: Ah yeah, indeed. The caller will still have the original `len`:
https://play.golang.org/p/DADY4bDN8eb

Comment (GerardRodes) [reply sliceutil/sliceutil.go]: yeah, you are right, better mutate it

Comment (rafaeljusto) [reply sliceutil/sliceutil.go]: You could just call the FilterString function now, as the implementation looks similar 😄 
```go
func RemoveString(list []string, s string) ([]string) {
  return FilterString(list, func(v string) bool {
    return s != v
  })
}
```

Comment (GerardRodes) [reply sliceutil/sliceutil.go]: Yes, I thought about it, I event just thought into deleting the entire function

But I do not like it that much the callback approach, it gives me JS nightmares, but it's totally a personal taste

**Review by ripexz — APPROVED** [review]

**Pull request merged**